### PR TITLE
Update BasketMiddleware.get_basket() to show warning only if items have been merged

### PR DESCRIPTION
**Pull request by gwaidacher**
Show warning only if merged baskets had  items, including the total number of items that have been merged.

### CODE REVIEW

**Comment by gwaidacher** [comment]
The new "Details" of "Tests / lint_python" do not show the error including the line in the file.
Please help with debugging

**Comment by specialunderwear** [comment]
Try "make black"

**Comment by gwaidacher** [comment]
> Try "make black"

Thank you, it worked!

File: src/oscar/apps/basket/middleware.py [hunk 1]
```diff
@@ -176,13 +180,14 @@ def get_basket(self, request):
...
-        if num_baskets_merged > 0:
+        if num_items_merged > 0:
+            # show warning only if items have been merged
             messages.add_message(
                 request,
                 messages.WARNING,
                 _(
-                    "We have merged a basket from a previous session. Its contents "
-                    "might have changed."
+                    f"We have merged {num_items_merged} items from a "
```
Comment (specialunderwear) [inline src/oscar/apps/basket/middleware.py:189]: Unfortunately f strings don't work in translatable strings yet, please use %(name)s like it's done in other places.

**Review by specialunderwear — CHANGES_REQUESTED** [review]
Could you add a test that shows this works as intended?

**Comment by gwaidacher** [comment]
> Could you add a test that shows this works as intended?

I'll try next week; I think I should go this way:

1) add a product to the catalogue
2) add a registered user 
2) call AddToBasketForm() and set the product in form data as anonymous user (WebTestCase)
3) call login-form, login as registered user
4) get context from login (response = form.submit().follow())
5) check response.context['messages']

**Comment by specialunderwear** [comment]
@gwaidacher why are you always pusing each separate commit? We get a notification of each commit you push, which is a problem with regards to spam. Why don't you finish your feature completely on your computer and then only push when you are done and want to see the test results?

**Comment by gwaidacher** [comment]
> @gwaidacher why are you always pusing each separate commit? We get a notification of each commit you push, which is a problem with regards to spam. Why don't you finish your feature completely on your computer and then only push when you are done and want to see the test results?

Hi! Sorry, my test is already working in `<our_app>/basket/tests.py` - but I'm unable to get it running in `tests/functional/basket/`
``` 
./manage.py test <our_app>.basket
...
Ran 1 test in 0.900s
OK
```

**Comment by gwaidacher** [comment]
> @gwaidacher why are you always pusing each separate commit? We get a notification of each commit you push, which is a problem with regards to spam. Why don't you finish your feature completely on your computer and then only push when you are done and want to see the test results?

done, finally! please have a look. I expect tests run always in English, so translating the expected message (test_basket.py, line 114) is not necessary, right?

**Review by samar-hassan — APPROVED** [review]

**Review by samar-hassan — APPROVED** [review]

**Review by specialunderwear — APPROVED** [review]

**Pull request merged**